autogpt_platform/backend/linter.py

```diff
@@ -21,23 +21,31 @@ def run(*command: str) -> None:
         )
     except subprocess.CalledProcessError as e:
         print(e.output.decode("utf-8"), file=sys.stderr)
+        raise
 
 
 def lint():
-    try:
-        run("ruff", "check", *TARGET_DIRS, "--exit-zero")
-        run("ruff", "format", "--diff", "--check", LIBS_DIR)
-        run("isort", "--diff", "--check", "--profile", "black", BACKEND_DIR)
-        run("black", "--diff", "--check", BACKEND_DIR)
-        run("pyright", *TARGET_DIRS)
-    except subprocess.CalledProcessError as e:
-        print("Lint failed, try running `poetry run format` to fix the issues: ", e)
-        raise e
+    lint_step_args: list[list[str]] = [
+        ["ruff", "check", *TARGET_DIRS, "--exit-zero"],
+        ["ruff", "format", "--diff", "--check", LIBS_DIR],
+        ["isort", "--diff", "--check", "--profile", "black", BACKEND_DIR],
+        ["black", "--diff", "--check", BACKEND_DIR],
+        ["pyright", *TARGET_DIRS],
+    ]
+    lint_error = None
+    for args in lint_step_args:
+        try:
+            run(*args)
+        except subprocess.CalledProcessError as e:
+            lint_error = e
+
+    if lint_error:
+        print("Lint failed, try running `poetry run format` to fix the issues")
+        raise lint_error
 
 
 def format():
     run("ruff", "check", "--fix", *TARGET_DIRS)
     run("ruff", "format", LIBS_DIR)
     run("isort", "--profile", "black", BACKEND_DIR)
     run("black", BACKEND_DIR)
-    run("pyright", *TARGET_DIRS)
```

Comment:
Please don't remove this, it will be massive annoyance to keep running stuff separately for devs :'(

Reply:
Oh we have lint nvm haha